nautobot/utilities/tests/test_filters.py

```diff
@@ -769,3 +770,89 @@ def test_device_comments_multiple_value_charfield_iregex(self):
         params = {"comments__nire": ["^device"]}
         self.assertEqual(self.DeviceFilterSetWithComments(params, self.device_queryset).qs.count(), 0)
+
+
+class SearchFilterTest(TestCase):
+    """Tests for the `SearchFilter` filter class."""
+
+    filterset_class = SiteFilterSet
+
+    def setUp(self):
+
+        super().setUp()
+
+        self.region1 = Region.objects.create(name="Test Region 1", slug="test-region-1")
+        self.region2 = Region.objects.create(name="Test Region 2", slug="test-region-2")
+        self.site1 = Site.objects.create(region=self.region1, name="Test Site 1", slug="test-site1", asn=1234)
+        self.site2 = Site.objects.create(region=self.region2, name="Test Site 2", slug="test-site2", asn=12345)
+        self.site3 = Site.objects.create(region=None, name="Test Site 3", slug="test-site3")
+
+        self.queryset = Site.objects.all()
+
+    def get_filterset_count(self, params, klass=None):
+        """To save ourselves some boilerplate."""
+        if klass is None:
+            klass = self.filterset_class
+        return klass(params, self.queryset).qs.count()
+
+    def test_default_icontains(self):
+        """Test a default search for an "icontains" value."""
+        params = {"q": "Test Site"}
+        self.assertEqual(self.get_filterset_count(params), 3)
+        params = {"q": "Test Site 3"}
+        self.assertEqual(self.get_filterset_count(params), 1)
+        # Trailing space should also match all
+        params = {"q": "Test Site "}
+        self.assertEqual(self.get_filterset_count(params), 3)
+
+    def test_default_exact(self):
+        """Test a default search for an "exact" value."""
+        params = {"q": "1234"}
+        self.assertEqual(self.get_filterset_count(params), 1)
+        params = {"q": "123"}
+        self.assertEqual(self.get_filterset_count(params), 0)
+
+    def test_typed_valid(self):
+        """Test that validly-typed predicate mappings are handled correctly."""
+
+        class MySiteFilterSet(SiteFilterSet):
+            """Overload the default just to illustrate that it's all we're testing for here."""
+
+            q = SearchFilter(filter_predicates={"asn": {"lookup_expr": "exact", "preprocessor": int}})
+
+        params = {"q": "1234"}
+        self.assertEqual(self.get_filterset_count(params, MySiteFilterSet), 1)
+        params = {"q": "123"}
+        self.assertEqual(self.get_filterset_count(params, MySiteFilterSet), 0)
+
+        # Further an invalid type (e.g. dict) will just result in the query failing open.
+        class MySiteFilterSet2(SiteFilterSet):
+            """Overload the default just to illustrate that it's all we're testing for here."""
+
+            q = SearchFilter(filter_predicates={"asn": {"lookup_expr": "exact", "preprocessor": dict}})
+
+        params = {"q": "1234"}
+        self.assertEqual(self.get_filterset_count(params, MySiteFilterSet2), Site.objects.count())
```

Comment:
I really don't like that filters "fail open"  - here we have a SearchFilter that won't match the provided query string, and the return is all Sites rather than an empty list? Does this fall under #1736 or is there something specific in the way SearchFilter is implemented that could be changed here?